Command/TerytImportCommand.php

```diff
@@ -33,30 +40,97 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 1;
         }
 
+        $xmlParser = $this->createXmlParser();
+        $this->getProgressHelper()->start($output, filesize($xmlFile));
+
+        $this->importXmlFile($xmlParser, $xmlFile);
+
+        $this->flushAndClear();
+        $this->getProgressHelper()->finish();
+
+        return 0;
+    }
+
+    /**
+     * @return Parser
+     * @throws \Exception
+     */
+    private function createXmlParser()
+    {
         $xmlParser = new Parser();
-        $xmlParser->registerCallback(
+
+        return $xmlParser->registerCallback(
             '/teryt/catalog/row',
             $this->getNodeParserCallbackFunction()
         );
-        $xmlParser->parse(fopen($xmlFile, 'r'));
-
-        return 0;
     }
 
     /**
      * @return callable
      */
-    protected function getNodeParserCallbackFunction()
+    private function getNodeParserCallbackFunction()
     {
-        $om = $this->getContainer()->get('doctrine')->getManager();
-        $self = $this;
+        $counter = self::FLUSH_FREQUENCY;
 
-        return function (Parser $parser, \SimpleXMLElement $node) use ($om, $self) {
-            $converter = $self->getNodeConverter($node, $om);
-            $entity = $converter->convertToEntity();
-            $om->persist($entity);
-            $om->flush();
-            $om->clear();
+        return function (Parser $parser, SimpleXMLElement $node) use (&$counter) {
+            $this->convertNodeToPersistedEntity($node);
+            $this->updateProgressHelper();
+
+            $counter--;
+            if (!$counter) {
```

Comment:
Właśnie po to jest ta linijka - https://github.com/fsi-open/teryt-database-bundle/pull/12/files#diff-cba843907da8872e35b747372ec8a535R48